x/superfluid/keeper/internal/events/emit_test.go

```diff
@@ -0,0 +1,339 @@
+package events_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v10/app/apptesting"
+	"github.com/osmosis-labs/osmosis/v10/x/superfluid/keeper/internal/events"
+	"github.com/osmosis-labs/osmosis/v10/x/superfluid/types"
+)
+
+type SuperfluidEventsTestSuite struct {
+	apptesting.KeeperTestHelper
+}
+
+const (
+	addressString = "addr1---------------"
+	testDenomA    = "denoma"
+	testDenomB    = "denomb"
+	testDenomC    = "denomc"
+	testDenomD    = "denomd"
+)
+
+func TestSuperfluidEventsTestSuite(t *testing.T) {
+	suite.Run(t, new(SuperfluidEventsTestSuite))
+}
+
+func (suite *SuperfluidEventsTestSuite) TestEmitSetSuperfluidAssetEvent() {
+	testcases := map[string]struct {
+		ctx       sdk.Context
+		denom     string
+		assetType types.SuperfluidAssetType
+	}{
+		"basic valid": {
+			ctx:       suite.CreateTestContext(),
+			denom:     testDenomA,
+			assetType: types.SuperfluidAssetTypeNative,
+		},
+		"context with no event manager": {
+			ctx: sdk.Context{},
+		},
```

Comment:
@p0mvn is there no need to test an invalid case, such as an invalid denom?

Reply:
I think this function already assumes that a denom is pre-validated somewhere else. As far as it is concerned, any received denom string is valid. So I think it should be fine without extra tests unless you would like to suggest anything specific that I'm not seeing.

Reply:
No that all sounds good to me, was just checking!